src/dsp/fax/fax_store.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * fax decoded lines sink block header
+ *
+ * Copyright 2022 Marc CAPDEVILLE F4JMZ
+ *
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3, or (at your option)
+ * any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#ifndef FAX_STORE_H
+#define FAX_STORE_H
+
+#include <gnuradio/sync_block.h>
+#include <boost/circular_buffer.hpp>
+#include <string>
+#include <mutex>
+
+namespace gr {
+   namespace fax {
+      class fax_store : public gr::sync_block {
+         public:
+
+#if GNURADIO_VERSION < 0x030900
+            typedef boost::shared_ptr<fax_store> sptr;
+#else
+            typedef std::shared_ptr<fax_store> sptr;
+#endif
+
+            static sptr make(int len);
+
+            int get_data(unsigned char *out,unsigned int *len);
+            void reset();
+
+            int work (int noutput_items,
+                  gr_vector_const_void_star &input_items,
+                  gr_vector_void_star &output_items);
+
+         private:
+            fax_store(int len);
+            ~fax_store();
+
+            std::mutex d_mutex;
+
+            int d_len;
+            boost::circular_buffer<std::vector<unsigned char>> d_data;
```

Comment:
Gqrx does not use boost (except `boost::shared_ptr`, which is required to support GNU Radio 3.8, and will be dropped as soon as Gqrx stops supporting 3.8). I don't want to add back boost as a dependency, so this will need to be replaced with either something in the standard library, or else a GNU Radio circular buffer.